src/Tools/dotnet-monitor/Auth/ApiKeyAuthenticationHandler.cs

```diff
@@ -34,9 +29,17 @@ internal sealed class ApiKeyAuthenticationHandler : AuthenticationHandler<ApiKey
             "SHA", "SHA1", "System.Security.Cryptography.SHA1", "System.Security.Cryptography.HashAlgorithm", "MD5", "System.Security.Cryptography.MD5"
         };
 
-        public ApiKeyAuthenticationHandler(IOptionsMonitor<ApiKeyAuthenticationHandlerOptions> options, ILoggerFactory loggerFactory, UrlEncoder encoder, ISystemClock clock)
+        private readonly ApiAuthenticationOptions _apiAuthOptions;
+
+        public ApiKeyAuthenticationHandler(
+            IOptionsMonitor<ApiKeyAuthenticationHandlerOptions> options,
+            IOptionsSnapshot<ApiAuthenticationOptions> apiAuthOptions,
+            ILoggerFactory loggerFactory,
+            UrlEncoder encoder,
+            ISystemClock clock)
             : base(options, loggerFactory, encoder, clock)
         {
+            _apiAuthOptions = apiAuthOptions.Value;
```

Comment:
Can you add a comment regarding why you can get the value instead of getting the snasphot later? IIRC this is a transient service, but that's not immediately obvious.

Reply:
Correct, I'll add comments to explain this.